src/lib/units/setTime.js

```diff
@@ -0,0 +1,55 @@
+import { getParseRegexForToken, isoTimes }   from '../parse/regex';
+import { prepareConfig } from '../create/from-anything';
+import { copyConfig } from '../moment/constructor';
+import { parseFromStringAndFormat } from '../create/from-string-and-format';
+import { HOUR, MINUTE, SECOND, MILLISECOND } from '../units/constants';
+import { isValid } from '../create/valid';
+import getParsingFlags from '../create/parsing-flags';
+
+
+export function setTime(input, format) {
+
+    if (!this.isValid())
+    {
+        return this;
+    }
+
+    if (typeof input === 'number')
+    {
+        return this.startOf('day').add(input, 'm');
+    }
+
+    if (!format)
+    {
+        format = findIsoFormat(input);
+    }
+
+    if (!format) {
+        this.invalidFormat = true;
+    }
+
+    this._f = format;
+    this._i = input;
+    this._isValid = null;
+
+    parseFromStringAndFormat(this);
```

Comment:
Seems like we should be working from a clone for this step instead of re-configuring the existing instance.